accepted/2020/net5/net5.md

```diff
@@ -0,0 +1,790 @@
+
+```C#
+public static class GpsLocation
+{
+    public static bool IsSupported
+    {
+        get
+        {
+#if ANDROID || IOS || WINDOWS
+            return true;
+#else
+            return false;
+#endif
+        }
+    }
+
+    public static (double Latitude, double Longitude) GetCoordinates()
+    {
+#if ANDROID
+        return AndroidAPI();
+#elif IOS
+        return AppleAPI();
+#elif WINDOWS
+        return WindowsAPI();
+#else
+        throw new PlatformNotSupportedException();
+#endif
+    }
+}
+```
+
+### Vary API surface
+
+Ada is a developer on SkiaSharp, a cross-platform 2D graphics API for .NET
+platforms based on Google's Skia Graphics Library. The project is already using
+multi-targeting to provide different implementations for different platforms. To
+make it easier to use, she's adding a new `SkiaSharpImage` type, which
+represents a bitmap and is constructed via OS-provided data types. Ada uses
+`#if` to expose different constructors on different platforms:
+
+```C#
+public static class SkiaSharpImage
+{
+#if ANDROID
+    public SkiaSharpImage(Android.Media.Image nativeImage) { /* ... */  }
+#endif
+
+#if IOS
+    public SkiaSharpImage(NSImage nativeImage) { /* ... */ }
+#endif
+
+#if WINDOWS
+    public SkiaSharpImage(Windows.Media.BitmapImage nativeImage) { /* ... */ }
+#endif
+}
+```
+
+### Upgrading the OS bindings
+
+Miguel is building Baby Shark, a popular iOS application. He started with .NET
+that supported iOS 13 but Apple just released iOS 14. He downloads the updated
+version of the .NET 5 SDK which now also includes support for iOS 14. In order
+to gain access to the new APIs that Apple has added, Miguel opens his project
+file which currently looks like this:
+
+```XML
+<Project Sdk="Microsoft.NET.Sdk">
+
+  <PropertyGroup>
+    <TargetFramework>net5.0-ios13.0</TargetFramework>
+  </PropertyGroup>
+
+  ...
+
+</Project>
+```
+
+He modifies the `<TargetFramework>` to be `net5.0-ios14.0`.
+
+### Lighting up on later OS versions
+
+Miguel doesn't want to cut off his users who are currently on iOS 13, so he
+wants to continue to have his application work on iOS 13 as well. To achieve
+that, Miguel modifies the project file by adding
+`<TargetPlatformMinimumVersion>`:
+
+```XML
+<Project Sdk="Microsoft.NET.Sdk">
+
+  <PropertyGroup>
+    <TargetFramework>net5.0-ios14.0</TargetFramework>
+    <TargetPlatformMinimumVersion>13.0</TargetPlatformMinimumVersion>
+  </PropertyGroup>
+
+  ...
+
+</Project>
+```
+
+However, since Miguel also uses the new `NSFizBuzz` API that Apple added in iOS
+14, he also modifies his source code to check for the operating system version
+before calling it:
+
+```C#
+public void OnClick(object sender, EventArgs e)
+{
+    if (Environment.OSVersion.Version >= new Version(14, 0))
+    {
+        NSFizBuzz();  
+    }
+}
+```
+
+### Consuming a library with a higher `TargetPlatformMinimumVersion`
+
+After using `NSFizzBuzz` directly for a while, Miguel notices that these OS APIs
```

Comment:
The way target framework compatibility is defined right now, there's no `half-way` compatibility between framework and this changes that. 

We should consider this to be a separate concept from target framework

Reply:
Addressed in the meeting. 

TargetPlatformMinimumVersion is not part of the TargetFramework friendly name, only TargetPlatformVersion is.

Reply:
Exactly.